records_mover/logging.py

```diff
@@ -40,7 +63,15 @@ def set_stream_logging(name: str = 'records_mover',
     adjusted_level = _adjusted_log_level(level, name)
     logger = logging.getLogger(name)
     logger.setLevel(adjusted_level)
-    handler = logging.StreamHandler(stream=stream)
+    wrapper = SecretsRedactingLogStream(stream)
```

Comment:
The logging system is a bit complicated and I don't know all of the ins and outs-- but I wonder if this could be done with a Filter (editing the log records directly) attached to the records_mover logger itself. That could be done globally, so it would work even if people turn on logging themselves rather than calling set_stream_logging.

Reply:
Ah, thanks, can't even remember whether I had looked at that option.  I gave implementation a shot and found out some things.

The challenge here is that the logging.LogRecord object we're handed in a registered logging.Filter class is very undigested - you redact all of the inputs to the logger, not the string output.

In the known secret leaking case here (Redshift load exceptions), we'd need to look to see if log_record.exc_info is specified, and if it is, look into the type of exception passed as the second element of the tuple.  If it's a sqlalchemy.exc.ProgrammingError, we'd need to downcast it and then sweep through the 'params' argument to redact it.

Unfortunately, that only really fixes this one situation, and at that point we might as well just apply a bandaid to that one place, I guess.

Let me know if that makes sense to you.

Reply:
Ah, yeah, I hadn't thought about the need to stringify the exception before we could do anything. In that case, I wonder if the thing we actually want is the formatter, since that's where the stringification of exceptions (and everything else) happens. I think that would make this usable even if you use a different handler (e.g, use the rotating file handler instead of the stream handler). But that's a future improvement.